google-cloud-pubsub/lib/google/cloud/pubsub/async_publisher/batch.rb

```diff
@@ -178,19 +178,16 @@ def reset!
                 return false
               end
 
-              until @queue.empty?
-                item = @queue.first
-                added = try_add item.msg, item.callback
-                break unless added
-                @queue.shift
-              end
+              min_create_time = refill_items
 
               return false unless @publishing
               if @items.empty?
                 @publishing = false
                 return false
               else
                 return true if stopping?
+                return true if (Time.now - min_create_time > @publisher.interval) ||
+                               (total_message_count == @publisher.max_messages)
```

Comment:
I suggest moving this to a helper function, or assigning to a boolean variable to indicate what exactly this condition is checking for.